pkg/pgmodel/extension.go

```diff
@@ -20,6 +20,36 @@ var (
 	ExtensionIsInstalled = false
 )
 
+// checkVersions is responsible for verifying the version compatibility of installed Postgresql database and extensions.
+func checkVersions(conn *pgx.Conn) error {
+	if err := checkPgVersion(conn); err != nil {
+		return fmt.Errorf("pg-version: %w", err)
+	}
+	if err := checkExtensionsVersion(conn); err != nil {
+		return fmt.Errorf("extensions-version: %w", err)
+	}
+	return nil
+}
+
+func checkPgVersion(conn *pgx.Conn) error {
+	var versionString string
+	if err := conn.QueryRow(context.Background(), "SHOW server_version_num;").Scan(&versionString); err != nil {
```

Comment:
This should be "SHOW server_version;"  which will return a X.Y number like "12.2" -> convert to semver via "%s.0"
https://www.postgresql.org/support/versioning/